### PR TITLE
upload_sorta_working

### PR DESCRIPTION
**Pull request by franclin1**

### CODE REVIEW

File: infrastructure/setting_up_state_s3.sh [hunk 1]
```diff
@@ -0,0 +1,8 @@
+#Name of the state bucket
+TERRAFORM_S3_BUCKET_NAME=capstone-bucket-malte-cgn-tf-state
+if aws s3 ls "s3://$TERRAFORM_S3_BUCKET_NAME" 2>&1 | grep -q 'An error occurred'
+then
+    aws s3api create-bucket --bucket $TERRAFORM_S3_BUCKET_NAME --region eu-central-1 --create-bucket-configuration LocationConstraint=eu-central-1
+else
+    echo "Terraform bucket to store state file exists allready"
```
Comment (T-Kutzner) [inline infrastructure/setting_up_state_s3.sh]:
```suggestion
    echo "Terraform bucket to store state file exists already"
```

**Review by T-Kutzner — APPROVED** [review]
only a little typo ;)

File: infrastructure/iam_roles.tf [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+    Version = "2012-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Sid    = ""
+        Principal = {
+          Service = "ecs-tasks.amazonaws.com"
+        }
+      },
```
Comment (T-Kutzner) [inline infrastructure/iam_roles.tf]:
```suggestion
      }
```

File: infrastructure/iam_roles.tf [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+                "ecr:GetAuthorizationToken",
+                "ecr:BatchCheckLayerAvailability",
+                "ecr:GetDownloadUrlForLayer",
+                "ecr:BatchGetImage",
+                "logs:CreateLogStream",
+                "logs:PutLogEvents"
+            ],
+        Effect   = "Allow",
+        Resource = "*"
+      },
```
Comment (T-Kutzner) [inline infrastructure/iam_roles.tf]:
```suggestion
      }
```

File: infrastructure/iam_roles.tf [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+    Version = "2008-10-17"
+    Statement = [
+      {
+        Action = "sts:AssumeRole"
+        Effect = "Allow"
+        Sid    = ""
+        Principal = {
+          Service = "ecs-tasks.amazonaws.com"
+        }
+      },
```
Comment (T-Kutzner) [inline infrastructure/iam_roles.tf]:
```suggestion
      }
```

**Review by T-Kutzner — APPROVED** [review]
lgtm, think the commas not necessary

File: infrastructure/fargate_image_upload.tf [hunk 5]
```diff
@@ -0,0 +1,90 @@
...
+
+resource "docker_registry_image" "endpoint_image" {
+    name = "${aws_ecr_repository.receipt_ident_repository.repository_url}"
+
+          build {
+          platform = "linux/amd64"
+          context = "../python"
+          dockerfile = "Dockerfile"
+    
+}
```
Comment (fabianschmauder) [inline infrastructure/fargate_image_upload.tf:18]: please format the file

File: infrastructure/fargate_image_upload.tf [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+
+resource "aws_ecs_cluster" "receipt_ident_cluster" {
+  name = "receipt_ident_cluster"
+
+  setting {
+    name  = "containerInsights"
+    value = "enabled"
+  }
+}
+resource "aws_ecs_cluster" "example" {
```
Comment (fabianschmauder) [inline infrastructure/fargate_image_upload.tf:29]: rename exmple

File: python/main.py [hunk 7]
```diff
@@ -0,0 +1,28 @@
...
+
+
+app = FastAPI()
+templates = Jinja2Templates(directory="htmldirectory")
+
+
+
+## file upload
+@app.post("/submit")
+async def root(receipt_image: UploadFile = File(...)):
```
Comment (fabianschmauder) [inline python/main.py:17]: i would rename the function. 
and maybe use another path like "/upload" or "/image"

Comment (franclin1) [reply python/main.py:17]: changed

File: infrastructure/fargate_image_upload.tf (hunk 6 above)
Comment (franclin1) [reply infrastructure/fargate_image_upload.tf:29]: done

File: infrastructure/fargate_image_upload.tf (hunk 5 above)
Comment (franclin1) [reply infrastructure/fargate_image_upload.tf:18]: done aswell

**Pull request merged**